### PR TITLE
Add retail customization function

### PR DESCRIPTION
**Pull request by gcandal**
Adds a function to ease customization and common assertions based on it.

### CODE REVIEW

File: src/ripe_rainbow/domain/base/waits.py [hunk 1]
```diff
@@ -6,15 +6,34 @@
...
     def __init__(self, owner):
         parts.Part.__init__(self, owner)
         self.wait = WebDriverWait(self.driver, self.timeout)
 
+    def wrap_method(self, method, *args):
+        """
+        Wraps the method being waited on to tolerate some exceptions since in most cases these are transitory conditions
+        that shouldn't break the test.
+
+        :param method: The method being ran.
```
Comment (joamag) [inline src/ripe_rainbow/domain/base/waits.py]: add the :type and :rtype

File: src/ripe_rainbow/domain/logic/retail.py [hunk 2]
```diff
@@ -55,6 +55,57 @@ def select_size(self, scale, size):
         self.interactions.click_when_possible(".size .button.button-primary.button-apply")
         self.waits.is_not_visible(".size .modal-container")
 
+    def change_customization(self, brand, model, part, material, color):
+        """
+        Makes a change to the customization of a part and checks that the pages mutates correctly, picking the right
+        active parts, materials and colors, as well as properly switching the swatches.
+
+        :param brand: The brand of the model.
```
Comment (joamag) [inline src/ripe_rainbow/domain/logic/retail.py]: :type and :rtype and try to break the lines according to the typical line width

Comment (gcandal) [reply src/ripe_rainbow/domain/logic/retail.py]: What's our approximate line width? I was using PEP-8 max width as reference.

File: src/ripe_rainbow/domain/logic/retail.py [hunk 3]
```diff
@@ -55,6 +55,62 @@ def select_size(self, scale, size):
...
+
+        self.waits.text(".button-material.active", material.upper())
+
+        self.waits.text(".button-color.active", color.capitalize())
+        self.waits.until(
+            lambda d: self.swatch_is_correct(".pickers .button-color.active .swatch > img", brand, model, material, color),
+            "Color swatch didn't have the expected image."
+        )
+
+    def swatch_is_correct(self, selector, brand, model, material, color):
```
Comment (joamag) [inline src/ripe_rainbow/domain/logic/retail.py]: assert_swatch

File: src/ripe_rainbow/domain/logic/retail.py [hunk 4]
```diff
@@ -55,6 +55,62 @@ def select_size(self, scale, size):
         self.interactions.click_when_possible(".size .button.button-primary.button-apply")
         self.waits.is_not_visible(".size .modal-container")
 
+    def change_customization(self, brand, model, part, material, color):
```
Comment (joamag) [inline src/ripe_rainbow/domain/logic/retail.py]: use the `set_part` naming

**Pull request merged**